File: 2. Programming Fundamentals with Java/Methods/PasswordValidator.java
import java.util.Scanner;

public class PasswordValidator {
    public static void main(String[] args) {

        passwordValidator();
    }

    public static void passwordValidator(){
        Scanner scanner = new Scanner(System.in);
        String password = scanner.nextLine();

        if (password.length() < 6 || password.length() > 10){
            System.out.println("Password must be between 6 and 10 characters");
        }
        if (!password.matches("[a-zA-Z0-9]*")){
            System.out.println("Password must consist only of letters and digits");
        }
        if (!password.matches("^(?=.*?\\d.*\\d)[a-zA-Z0-9]{8,}$")){
            System.out.println("Password must have at least 2 digits");
        }else {
            System.out.println("Password is valid");
        }
    }
}

File: 9. JS Front-End/Functions and Statements/PasswordValidator.js
function passValidator(password) {
    const isValidLenght = (pass) => pass.length >= 6 && pass.length <= 10;
    const hasOnlyLettersAndDigits = (pass) => /^[A-Za-z0-9]*$/.test(pass);
    const hsaMoreThanTwoDigits = (pass) => [...pass.matchAll(/\d/g)].length >= 2;

    let passIsValid = true;

    if (!isValidLenght(password)) {
        console.log("Password must be between 6 and 10 characters");
        passIsValid = false;
    }

    if (!hasOnlyLettersAndDigits(password)) {
        console.log("Password must consist only of letters and digits");
        passIsValid = false;
    }

    if (!hsaMoreThanTwoDigits(password)) {
        console.log("Password must have at least 2 digits");
        passIsValid = false;
    }

    if (passIsValid) {
        console.log("Password is valid")
    }
}

passValidator('Pa$s$s');